Core/NoOpRunner.Core/Interfaces/IObserver.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace NoOpRunner.Core.Interfaces
+{
+    public interface IObserver
+    {
+        void Update(object sender, object args);
```

Comment:
Why use objects? You are doing way too much upcasting in this impl, it's smelly. Sender can definitely be more narrow, args probably as well.